prime-angular-frontend/src/app/modules/enrolment/pages/care-setting/care-setting.component.ts

```diff
@@ -223,28 +222,23 @@ export class CareSettingComponent extends BaseEnrolmentProfilePage implements On
 
     switch (careSettingCode) {
       case CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE: {
-        communityHealthSites.reset();
+        communityHealthSites.clear();
```

Comment:
It might be worth creating a method that invokes those 3 methods on the passed in FormArray to reduce a bit of repetition.

Reply:
Also, I noticed you loop over the `oboSites.controls`, and while you're doing it you're removing an index from the list you're iterating over so the list is changing in size while you loop.  This can lead to some very weird bugs.  I would loop over the list using a reduce and collect a list of indices you want to remove, and then I would remove them in a chained `forEach`.